modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -1143,45 +1297,20 @@ bool CvCapture_FFMPEG::processRawPacket()
     {
         if (packet_filtered.data)
         {
-            _opencv_ffmpeg_av_packet_unref(&packet_filtered);
-        }
-
-#if LIBAVFORMAT_BUILD >= CALC_FFMPEG_VERSION(58, 20, 100)
-        int err = av_bsf_send_packet(bsfc, &packet);
-        if (err < 0)
-        {
-            CV_WARN("Packet submission for filtering failed");
-            return false;
-        }
-        err = av_bsf_receive_packet(bsfc, &packet_filtered);
-        if (err < 0)
-        {
-            CV_WARN("Filtered packet retrieve failed");
-            return false;
+            compat::avPacketUnref(&packet_filtered);
         }
-#else
-#if LIBAVFORMAT_BUILD > 4628
-        AVCodecContext* ctx = ic->streams[video_stream]->codec;
-#else
-        AVCodecContext* ctx = &ic->streams[video_stream]->codec;
-#endif
-        int err = av_bitstream_filter_filter(bsfc, ctx, NULL, &packet_filtered.data,
-            &packet_filtered.size, packet.data, packet.size, packet_filtered.flags & AV_PKT_FLAG_KEY);
-        if (err < 0)
+        if (!compat::avBitstreamFilterFilter(bsfc, ic->streams[video_stream]->codec, packet,
+                                             packet_filtered))
```

Comment:
Avoid extraction of one time used code into functions.

Reply:
I tried to remove versions `#ifdef`'s from the actual functions of the capture and writer, to keep them simple and clean. What is the matter to do not do it?